node/pkg/logscribe/app.go

```diff
@@ -87,3 +163,133 @@ func bulkCopyLogs(ctx context.Context, logsChannel <-chan *[]api.LogInsertModel)
+		topOccurrences = topOccurrences[:topOccurrencesForGithubIssue]
+	}
+
+	for _, pair := range topOccurrences {
+		processedLogs = append(
+			processedLogs,
+			LogInsertModelWithIDWithCount{OccurrenceCount: pair.count, LogInsertModelWithID: pair.log},
+		)
+	}
+
+	return processedLogs
+}
+
+func buildLogMap(logs []LogInsertModelWithID) map[string]LogsWithCount {
+	logHashCountMap := make(map[string]LogsWithCount) // {"hashedLog: {count, log}}
+	for _, log := range logs {
+		hash := hashLog(log)
+		logsWithCount, exists := logHashCountMap[hash]
+		if !exists {
+			logsWithCount = LogsWithCount{
+				count: 0,
+				log:   log,
+			}
+		}
+		logsWithCount.count++
+		logHashCountMap[hash] = logsWithCount
+	}
+	return logHashCountMap
+}
+
+func getTopOccurrences(hashLogPairs map[string]LogsWithCount) []LogsWithCount {
+	pairs := make([]LogsWithCount, 0, len(hashLogPairs))
+	for _, pair := range hashLogPairs {
+		pairs = append(pairs, pair)
+	}
+	sort.Slice(pairs, func(i, j int) bool {
+		return pairs[i].count > pairs[j].count
+	})
+	return pairs
+}
+
+func fetchDeleteLogs(ctx context.Context, service string) ([]LogInsertModelWithID, error) {
+	logs, err := db.QueryRows[LogInsertModelWithID](ctx, readDeleteLogsQuery, map[string]any{
+		"service": service,
+	})
+	if err != nil {
+		log.Error().Err(err).Msg("Failed to read logs")
+		return nil, err
+	}
+	return logs, nil
+}
+
+func hashLog(log LogInsertModelWithID) string {
+	hash := sha256.New()
+	hash.Write([]byte(log.Service))
+	hash.Write([]byte(fmt.Sprintf("%d", log.Level)))
+	hash.Write([]byte(log.Message))
+	hash.Write(log.Fields)
+	return hex.EncodeToString(hash.Sum(nil))
+}
+
+func (a *App) createGithubIssue(ctx context.Context, processedLogs []LogInsertModelWithIDWithCount, service string) {
+	if a.githubOwner == "test" {
+		return
+	}
+
+	issueCount := 0
+	processedLogHashes := [][]interface{}{}
+	for _, entry := range processedLogs {
+		logsJson := ""
+		hash := hashLog(entry.LogInsertModelWithID)
+		res, err := db.QueryRow[Count](ctx, logAlreadyProcessedQuery, map[string]any{
+			"hash": hash,
+		})
+		if err != nil {
+			log.Error().Err(err).Msg("Failed to check if log already processed")
+			continue
+		}
+		if res.Count > 0 {
+			log.Debug().Msg("Log already processed, skipping creation of github issue")
+			continue
+		}
+
+		entryJson, err := json.MarshalIndent(entry, "", "  ")
+		if err != nil {
+			log.Error().Err(err).Msg("Failed to marshal log")
+			return
+		}
+		logsJson += string(entryJson) + "\n"
+		formattedBody := "```go\n" + logsJson + "\n```"
+		issueRequest := &github.IssueRequest{
+			Title:  github.String(fmt.Sprintf("[%s] %s", service, entry.Message)),
+			Body:   github.String(formattedBody),
+			Labels: &[]string{"bug"},
+		}
+
+		_, resp, err := a.githubClient.Issues.Create(ctx, a.githubOwner, a.githubRepo, issueRequest)
+		if err != nil || resp.StatusCode != http.StatusCreated {
+			log.Error().Err(err).Msg("Failed to create github issue")
+		}
```

Comment:
I'm concerned if it might have some kind of rate limit, might be safer to wrap with retrier or sleep for a sec at the end of the iteration